util/updateXML/sources/Model/Data/Model.py

```diff
@@ -0,0 +1,194 @@
+from abc import ABC
+import lxml.etree
+
+from ...Common import *
+from ...UpdateXMLExceptions import *
+
+
+class Model(ABC):
+    """
+    This abstract class defines a common interface for classes which own a parset to :
+        - add parameters/references in parset
+        - remove parameters/references from parset
+        - change parameters/references name
+
+    Attribute
+    ----------
+    parset_ : str
+        parset content
+    """
+    def __init__(self, parset):
+        self.parset_ = parset
+
+    # ---------------------------------------------------------------
+    #   USER METHODS
+    # ---------------------------------------------------------------
+
+    def add_param(self, param_type, param_name, param_value):
+        """
```

Comment:
This should be in Par.py

Reply:
The methods to modify parsets are called with BlackBoxModel, Network, Solver and UnitDynamicModel classes. Not Par class.

Reply:
I don't understand the logic: why having an almost empty par class and put parameter related methods somewhere else?

Reply:
The Par class is just a wrapper to parse a Par file and easily get its filename when we output the modified files.

Then, when we modify the parset (in BlackBoxModel, Network, Solver and UnitDynamicModel) we just need the parset related to our object, not the whole Par file.

Today, we use classes which inherite from the Model class to modify the parsets.

Reply:
I disagree with this as data and method to modify them are put in different places. The blackbox and others should contain an instance of the par class, that itself contains the methods to modify the parameters and references

Reply:
The Parset class is actually the Model class

Reply:
then please name it ParSet